tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/ServerlessMiniAgent.cs

```diff
@@ -0,0 +1,113 @@
+// <copyright file="ServerlessMiniAgent.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.IO;
+
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.ServerlessInstrumentation;
+
+internal class ServerlessMiniAgent
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ServerlessMiniAgent));
+    internal const string AzureFunctionNameEnvVar = "WEBSITE_SITE_NAME";
+    internal const string AzureFunctionIdentifierEnvVar = "FUNCTIONS_EXTENSION_VERSION";
+    internal const string GCPFunctionDeprecatedNameEnvVar = "FUNCTION_NAME";
+    internal const string GCPFunctionDeprecatedEnvVarIdentifier = "GCP_PROJECT";
+    internal const string GCPFunctionNewerNameEnvVar = "K_SERVICE";
+    internal const string GCPFunctionNewerEnvVarIdentifier = "FUNCTION_TARGET";
+
+    internal static bool IsGCPFunction { get; private set; } = GetIsGCPFunction();
+
+    internal static bool IsAzureFunction { get; private set; } = GetIsAzureFunction();
+
+    internal static void MaybeStartMiniAgent(ServerlessMiniAgentManager manager)
```

Comment:
You pass `manager` here, but you only use it in the last line: `manager.Start(rustBinaryPath)`. I think we could avoid some coupling between these two types if we change this signature to:
```diff
- internal static void MaybeStartMiniAgent(ServerlessMiniAgentManager manager)
+ internal static string GetBinaryPath()
```

and change the last line to:
```diff
- manager.Start(rustBinaryPath);
+ return rustBinaryPath;
```

And then maybe rename this class to `ServerlessMiniAgentPathHelper` or something along those lines, since that it all it does :)

Reply:
fixed. I ended up refactoring the two files, combining them into a single `ServerlessMiniAgent` file. I figured since both classes are static, and the ServerlessMiniAgentManager wasn't doing any managing aside from the single spawn function, it would make more sense to combine everything.